uwblueprint/building-up-client
# Fetch user data on page load to store globally in Redux store

### Description
Note: backend ticket must be merged first (https://github.com/uwblueprint/building-up-server/pull/32)
- Updated login mutation and current user query to fetch additional user data (teamId, userId, lastName)
- Logging in will automatically update Redux store with user data, but this will be lost once the user refreshes the page
- `CURRENT_USER` query uses cookie (containing token with userId) to query backend for user data on page load/refresh (if they're logged in)
- Created/modified ProtectedRoute component (#23) to add UI for logging in/out, for testing purposes

### Screenshots
#### Logging in
![2021-02-08 20 09 13](https://user-images.githubusercontent.com/44102730/107303032-46093780-6a4c-11eb-91f2-d2aba88ec489.gif)

#### Page refresh
![2021-02-08 20 10 22](https://user-images.githubusercontent.com/44102730/107303040-4bff1880-6a4c-11eb-8717-f1b2fed87411.gif)

#### Logging out
![2021-02-08 20 10 41](https://user-images.githubusercontent.com/44102730/107303046-4f929f80-6a4c-11eb-8471-88e30f834e7a.gif)

### How to Test
- Try logging in, refreshing page as a logged in user, and logging out, and make sure the Redux state is correct
